src/core/reactor_backend.cc

```diff
@@ -182,6 +182,11 @@ aio_storage_context::handle_aio_error(linux_abi::iocb* iocb, int ec) {
             // we will only remove it from _pending_io and try again.
             return 1;
         }
+        case ENOTSUP:
+            throw_system_error_on(true,
+                                  "could not submit io. this happens when accessing "
+                                  "filesystem not support asynchronous direct I/O");
+            abort();
```

Comment:
why this abort?

Reply:
also, why throw? it can't be handled

Reply:
@avikivity i was following the approach of the `default` branch.  as i thought we should just abort the application in this case, as we might not a good way to handle this issue.

but after a second thought, it might be desirable to return this error to the caller as an `std::system_error`.